code/game/objects/items/stacks/medical.dm

```diff
@@ -10,8 +10,8 @@
 	var/heal_brute = 0
 	var/heal_burn = 0
 
-/obj/item/stack/medical/attack(mob/living/carbon/M, mob/living/user)
-	if (!istype(M))
+/obj/item/stack/medical/attack(mob/living/M, mob/living/user)
+	if (!istype(M) || istype(M, /mob/living/silicon))
```

Comment:
For this, use get_classification
It returns a bitfield which will be some combination of these 

https://i.imgur.com/DBgKIbT.png

you'll want to only allow things that are organic